src/lektorium/repo/local/repo.py

```diff
@@ -14,6 +14,43 @@
     SessionNotFound,
 )
 from .objects import Session, Site
+import collections.abc
+
+
+class FilteredDict(collections.abc.Mapping):
+    def __init__(self, keys, dct):
+        self.__keys = keys
+        self.__dct = dct
+
+    def __getitem__(self, key):
+        if key not in self.__keys:
+            raise KeyError(key)
+        return self.__dct[key]
+
+    @property
+    def __common_keys(self):
+        return set(self.__keys).intersection(self.__dct)
+
+    def __iter__(self):
+        return iter(self.__common_keys)
+
+    def __len__(self):
+        return len(self.__common_keys)
```

Comment:
What is the point of using dunder properties in this class?

Reply:
I am aware of clashing names with base abc.Mapping class coming from stdlib.